server/events/vcs/github_client.go

```diff
@@ -579,13 +579,23 @@ func (g *GithubClient) MergePull(pull models.PullRequest, pullOptions models.Pul
 	if err != nil {
 		return errors.Wrap(err, "fetching repo info")
 	}
+	protection, _, err := g.client.Repositories.GetBranchProtection(context.Background(), repo.Owner.GetLogin(), *repo.Name, pull.BaseBranch)
+	if err != nil {
+		if !errors.Is(err, github.ErrBranchNotProtected) {
+			return errors.Wrap(err, "getting branch protection rules")
```

Comment:
How come we want to do an early return here?

Reply:
We check if the branch is protected by calling the API endpoint.
If branch is protected we get back the response and no error.
If branch is not protected we get back nil response and specific error message.
All other cases we treat as standard communication errors 401, 504, 500 etc.